### PR TITLE
chore(sto-bro): add error message handling

### PR DESCRIPTION
**Pull request by HuiSF**
#### Description of changes

* generate `message` via callback retuned by `useDisplayText()`
* add `MessageControl` to each view

#### Description of how you validated changes

* unit test
* sample app test

#### Checklist

- [x] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [x] PR description included
- [x] `yarn test` passes and tests are updated/added
- [x] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [ ] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-storage/src/components/StorageBrowser/views/LocationActionView/CreateFolderView/__tests__/CreateFolderView.spec.tsx [hunk 1]
```diff
@@ -156,7 +156,7 @@ describe('CreateFolderView', () => {
       data: {
         isActionStartDisabled: true,
         folderNameId,
-        messageContent: 'Success!',
+        message: 'Success!',
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageBrowser/views/LocationActionView/CreateFolderView/__tests__/CreateFolderView.spec.tsx:159]: This is the wrong shape for `message`, we must be covering up the TS error with a cast somewhere above maybe?

File: packages/react-storage/src/components/StorageBrowser/views/LocationActionView/DeleteView/DeleteView.tsx [hunk 2]
```diff
@@ -69,9 +83,7 @@ export function DeleteView({
           <StatusDisplayControl />
         </ViewElement>
         <ViewElement className={`${STORAGE_BROWSER_BLOCK}__footer`}>
-          <ViewElement className={`${STORAGE_BROWSER_BLOCK}__message`}>
-            {/* TODO: confirmation message goes here */}
-          </ViewElement>
+          <MessageControl />
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageBrowser/views/LocationActionView/DeleteView/DeleteView.tsx:86]:
```suggestion
          <ViewElement className={`${STORAGE_BROWSER_BLOCK}__message`}>
            <MessageControl />
          </ViewElement>
```

File: packages/react-storage/src/components/StorageBrowser/views/LocationActionView/UploadView/UploadView.tsx [hunk 3]
```diff
@@ -142,7 +150,7 @@ export function UploadView({
           <StatusDisplayControl />
         </ViewElement>
         <ViewElement className={`${STORAGE_BROWSER_BLOCK}__footer`}>
-          {/* Message goes here */}
+          <MessageControl />
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageBrowser/views/LocationActionView/UploadView/UploadView.tsx:153]:
```suggestion
          <ViewElement className={`${STORAGE_BROWSER_BLOCK}__message`}>
            <MessageControl />
          </ViewElement>
```

File: packages/react-storage/src/components/StorageBrowser/views/LocationDetailView/LocationDetailView.tsx [hunk 4]
```diff
@@ -91,6 +73,15 @@ export function LocationDetailView({
...
   } = useLocationDetailView({ onNavigate: onNavigateProp, onExit });
 
+  const messageControlContent = hasError
+    ? getListItemsResultMessage({
+        items: pageItems,
+        errorMessage: message ?? DEFAULT_ERROR_MESSAGE,
+      })
+    : getListItemsResultMessage({
+        items: pageItems,
+      });
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageBrowser/views/LocationDetailView/LocationDetailView.tsx:83]: Can drop `DEFAULT_ERROR_MESSAGE`, that should be returned from the callback. We should also provide `query` for deriving search specific messages:

```suggestion
  const messageControlContent = getListItemsResultMessage({
    items: pageItems,
    errorMessage: message,
    query: searchQuery,
  });
```

We may need to update `getListItemsResultMessage` to return `undefined` by default

File: packages/react-storage/src/components/StorageBrowser/views/LocationsView/LocationsView.tsx [hunk 5]
```diff
@@ -119,6 +102,16 @@ export function LocationsView({
...
 
+  // TODO: add hasExhaustedSearch param
+  const messageControlContent = hasError
+    ? getListLocationsResultMessage({
+        locations: pageItems,
+        errorMessage: message ?? DEFAULT_ERROR_MESSAGE,
+      })
+    : getListLocationsResultMessage({
+        locations: pageItems,
+      });
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageBrowser/views/LocationsView/LocationsView.tsx:113]: Same comment as https://github.com/aws-amplify/amplify-ui/pull/6096/files#diff-80fb0028a28c05f04c78c596fdd93ec6dfc4c653d8eb1a233210c10480bd3422R83

**Review by calebpollman — COMMENTED** [review]
Thanks for doing this! Couple small changes but other than that lgtm. We will need to do clean up of the actual returned messages but that can be done in a follow up

**Comment by AllanZhengYP** [comment]
Should we model messages as array to support multiple messages?

**Comment by calebpollman** [comment]
> Should we model messages as array to support multiple messages?

Not in the immediate term

**Review by AllanZhengYP — DISMISSED** [review]
Looks good to me!

File: packages/react-storage/src/components/StorageBrowser/displayText/libraries/en/locationDetailView.ts [hunk 6]
```diff
@@ -1,10 +1,45 @@
...
+        return {
+          type: 'info',
+          content: 'No files.',
+        };
+      }
+
+      if (query && hasExhaustedSearch) {
+        return {
+          type: 'warning',
+          content: `No more folders or files found by ${query}.`,
```
Comment (pranavosu) [inline packages/react-storage/src/components/StorageBrowser/displayText/libraries/en/locationDetailView.ts:35]:
```suggestion
          type: 'info',
          content: `Showing results for up to the first 10,000 items`,
```

We can remove `searchExhaustedMessage ` below.

File: packages/react-storage/src/components/StorageBrowser/views/LocationDetailView/LocationDetailView.tsx [hunk 7]
```diff
@@ -96,8 +75,29 @@ export function LocationDetailView({
...
+    messageControlContent = getListItemsResultMessage({
+      items: pageItems,
+      hasExhaustedSearch,
+      message,
+    });
+  } else {
+    messageControlContent = getListItemsResultMessage({
+      items: pageItems,
+    });
+  }
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageBrowser/views/LocationDetailView/LocationDetailView.tsx:99]: We should not be providing params to `getListItemsResults` conditionally. `getListItemsResults` should always be provided with all parameters and decide when to return a message, same comment for `LocationsView`. Can be updated in a follow up.

**Review by pranavosu — APPROVED** [review]

**Review by calebpollman — APPROVED** [review]
Left feedback to be addressed in a follow up, other than lgtm. Thanks @HuiSF

**Pull request merged**